packages/server/lib/browsers/memory/index.ts

```diff
@@ -160,84 +204,130 @@ export const gatherMemoryStats: () => Promise<void> = measure(async () => {
+  statsLog.shouldCollectGarbage = shouldCollectGarbage
   statsLog.timestamp = Date.now()
 }, { name: 'gatherMemoryStats', save: true })
 
+/**
+ * Collects garbage if needed and logs the test information.
+ * @param automation the automation client used to collect garbage
+ * @param test the current test
+ */
 const maybeCollectGarbageAndLog = async ({ automation, test }: { automation: Automation, test: { title: string, order: number, currentRetry: number }}) => {
-  await maybeCollectGarbage({ automation })
+  await maybeCollectGarbage(automation)
 
   gcLog.testTitle = test.title
   gcLog.testOrder = Number(`${test.order}.${test.currentRetry}`)
   gcLog.garbageCollected = collectGarbageOnNextTest
   gcLog.timestamp = Date.now()
 
   addCumulativeStats(gcLog)
+
+  // clear the flag so we don't collect garbage on every test
+  collectGarbageOnNextTest = false
 }
 
-const maybeCollectGarbage = measure(async ({ automation }: { automation: Automation }) => {
+/**
+ * Collects garbage if needed.
+ * @param automation the automation client used to collect garbage
+ */
+const maybeCollectGarbage: (automation: Automation) => Promise<void> = measure(async (automation) => {
   if (collectGarbageOnNextTest) {
     debug('forcing garbage collection')
     await automation.request('collect:garbage', null, null)
-    collectGarbageOnNextTest = false
   } else {
     debug('skipping garbage collection')
   }
 }, { name: 'maybeCollectGarbage', save: true })
 
+/**
+ * Adds the memory stats to the cumulative stats.
+ * @param stats memory stats to add to the cumulative stats
+ */
 const addCumulativeStats = (stats: { [key: string]: any }) => {
   debugVerbose('memory stats: %o', stats)
 
-  if (['1', 'true'].includes(process.env.CYPRESS_INTERNAL_SAVE_MEMORY_STATS as string)) {
+  if (['1', 'true'].includes(process.env.CYPRESS_INTERNAL_SAVE_MEMORY_STATS?.toLowerCase() as string)) {
     cumulativeStats.push(_.clone(stats))
   }
 
   stats = {}
 }
 
+/**
+ * Checks the memory usage to gather the stats and schedules the next check.
+ */
 const checkMemory = async () => {
-  await gatherMemoryStats()
-  addCumulativeStats(statsLog)
+  try {
+    await gatherMemoryStats()
+    addCumulativeStats(statsLog)
+  } catch (err) {
+    debug('error gathering memory stats: %o', err)
+  }
   scheduleMemoryCheck()
 }
 
+/**
+ * Schedules the next memory check based on the MEMORY_PROFILER_INTERVAL.
+ */
 const scheduleMemoryCheck = () => {
   if (started) {
     // not setinterval, since checkMemory is asynchronous
     timer = setTimeout(checkMemory, MEMORY_PROFILER_INTERVAL)
   }
 }
 
+/**
+ * Starts the memory profiler.
+ * @param automation the automation client used to interact with the browser
+ * @param spec the current spec file
+ */
 async function startProfiling (automation: Automation, spec: { fileName: string }) {
   if (started) {
     return
   }
 
   debugVerbose('start memory profiler')
 
+  // ensure we are starting from a clean state
   reset()
```

Comment:
In open mode, is this correctly reset when switching between e2e and component testing and a new browser is launched?

Reply:
Yes, it resets correctly when switching between modes.